src/h3/_cy/cells.pyx

```diff
@@ -0,0 +1,281 @@
+cimport h3lib
+from .h3lib cimport bool, int64_t, H3int
+from libc cimport stdlib
+
+from .util cimport (
+    check_cell,
+    check_res,
+    create_ptr,
+    create_mv,
+    empty_memory_view, # want to drop this import if possible
+)
+
+from .util import H3ValueError
+
+# todo: add notes about Cython exception handling
+
+
+# bool is a python type, so we don't need the except clause
+cpdef bool is_cell(H3int h):
+    """Validates an H3 cell (hexagon or pentagon)
+
+    Returns
+    -------
+    boolean
+    """
+    return h3lib.h3IsValid(h) == 1
+
+cpdef bool is_pentagon(H3int h):
+    return h3lib.h3IsPentagon(h) == 1
+
+cpdef int get_base_cell(H3int h) except -1:
+    check_cell(h)
+
+    return h3lib.h3GetBaseCell(h)
+
+
+cpdef int resolution(H3int h) except -1:
+    """Returns the resolution of an H3 Index
+    0--15
+    """
+    check_cell(h)
+
+    return h3lib.h3GetResolution(h)
+
+
+cpdef H3int parent(H3int h, res=None) except 1:
+    check_cell(h)
+
+    if res is None:
+        res = resolution(h) - 1
+    # todo: actually, do we want to raise an error if there are no children, or just return an empty set?
```

Comment:
todo sounds like it's applicable to the `children` operation - maybe delete?

Reply:
good catch! fixed.